tests/backup/backup_longevity_test.go

```diff
@@ -0,0 +1,409 @@
+	emailTriggerFunction = map[string]func(){
+		EmailReporter: TriggerEmailReporter,
+	}
+
+	BeforeEach(func() {
+		if !populateDone {
+			StartPxBackupTorpedoTest("BackupLongevityTest",
+				"Longevity Run For Backup", nil, 0, ATrivedi, Q1FY24)
+			populateIntervals()
+			//  populateDisruptiveTriggers()
+			populateDone = true
+		}
+	})
+
+	It("has to schedule app and introduce test triggers", func() {
+		log.InfoD("schedule apps and start test triggers")
+		watchLog := fmt.Sprintf("Start watch on K8S configMap [%s/%s]",
+			configMapNS, testTriggersConfigMap)
+
+		Step(watchLog, func() {
+			log.InfoD(watchLog)
+			watchConfigMap()
+		})
+
+		TriggerDeployBackupApps(&contexts, &triggerEventsChan)
+		TriggerAddBackupCluster(&contexts, &triggerEventsChan)
+		TriggerAddBackupCredAndBucket(&contexts, &triggerEventsChan)
+
+		var wg sync.WaitGroup
+		Step("Register test triggers", func() {
+			for triggerType, triggerFunc := range triggerFunctions {
+				log.InfoD("Registering trigger: [%v]", triggerType)
+				go backupEventTrigger(&wg, &contexts, triggerType, triggerFunc, &triggerLock, &triggerEventsChan)
+				wg.Add(1)
+			}
+		})
+		log.InfoD("Finished registering test triggers")
+		if Inst().MinRunTimeMins != 0 {
+			log.InfoD("Longevity Tests  timeout set to %d  minutes", Inst().MinRunTimeMins)
+		}
+
+		Step("Register email trigger", func() {
+			for triggerType, triggerFunc := range emailTriggerFunction {
+				log.InfoD("Registering email trigger: [%v]", triggerType)
+				go emailEventTrigger(&wg, triggerType, triggerFunc, &emailTriggerLock)
+				wg.Add(1)
+			}
+		})
+		log.InfoD("Finished registering email trigger")
+
+		CollectEventRecords(&triggerEventsChan)
+		wg.Wait()
+		close(triggerEventsChan)
+		Step("teardown all apps", func() {
+			for _, ctx := range contexts {
+				TearDownContext(ctx, nil)
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+		AfterEachTest(contexts)
+	})
+})
+
+func backupEventTrigger(wg *sync.WaitGroup,
+	contexts *[]*scheduler.Context,
+	triggerType string,
+	triggerFunc func(*[]*scheduler.Context, *chan *EventRecord),
+	triggerLoc *sync.Mutex,
+	triggerEventsChan *chan *EventRecord) {
+	defer wg.Done()
+
+	minRunTime := Inst().MinRunTimeMins
+	timeout := (minRunTime) * 60
+
+	start := time.Now().Local()
+	lastInvocationTime := start
+
+	for {
+		// if timeout is 0, run indefinitely
+		if timeout != 0 && int(time.Since(start).Seconds()) > timeout {
+			log.InfoD("Longevity Tests timed out with timeout %d  minutes", minRunTime)
+			break
+		}
+
+		waitTime, isTriggerEnabled := isTriggerEnabled(triggerType)
+
+		if isTriggerEnabled && time.Since(lastInvocationTime) > time.Duration(waitTime) {
+
+			triggerFunc(contexts, triggerEventsChan)
+			log.Infof("Trigger Function completed for [%s]\n", triggerType)
+
+			lastInvocationTime = time.Now().Local()
+
+		}
+		time.Sleep(controlLoopSleepTime)
+	}
+	os.Exit(0)
+}
+
+func emailEventTrigger(wg *sync.WaitGroup,
```

Comment:
please see if we can use emailEventTrigger in longevity test

Reply:
This is same as longevity test, we copied the function from same

Reply:
Hi @atrivedi-px I would like to understand if it is same can we use the existing function itself?